tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java

```diff
@@ -2278,31 +2266,62 @@ public static void test_json() throws Exception {
 
 		try (Statement stmt = conn.createStatement()) {
 			ResultSet rs = stmt.executeQuery("select [1, 5]::JSON");
-
 			rs.next();
-
 			JsonNode jsonNode = (JsonNode) rs.getObject(1);
-
 			assertTrue(jsonNode.isArray());
-
 			assertEquals(jsonNode.toString(), "[1, 5]");
 		}
 
 		try (Statement stmt = conn.createStatement()) {
 			ResultSet rs = stmt.executeQuery("select \'hello\'::JSON");
-
 			rs.next();
-
 			JsonNode jsonNode = (JsonNode) rs.getObject(1);
-
 			assertTrue(jsonNode.isString());
 			assertEquals(jsonNode.toString(), "hello");
 		}
 	}
 
+	public static void test_bug4218_prepare_types() throws Exception {
+		DuckDBConnection conn = (DuckDBConnection) DriverManager.getConnection("jdbc:duckdb:");
+		String query = "SELECT ($1 || $2)";
+		conn.prepareStatement(query);
+		assertTrue(true);
+	}
+
+	public static void test_bug532_timestamp() throws Exception {
+		Connection conn = DriverManager.getConnection("jdbc:duckdb:");
+		Statement stmt = conn.createStatement();
+
+		ResultSet rs;
+
+		stmt.execute("CREATE TABLE t0(c0 DATETIME);");
+		stmt.execute("INSERT INTO t0 VALUES(DATE '1-1-1');");
+		rs = stmt.executeQuery("SELECT t0.c0 FROM t0; ");
+
+		rs.next();
+		rs.getObject(1);
+	}
+
+	public static void test_bug966_typeof() throws Exception {
+		Connection conn = DriverManager.getConnection("jdbc:duckdb:");
+		Statement stmt = conn.createStatement();
+		ResultSet rs = stmt.executeQuery("select typeof(1);");
+
+		rs.next();
+		assertEquals(rs.getString(1), "INTEGER");
+	}
+
 	public static void main(String[] args) throws Exception {
 		// Woo I can do reflection too, take this, JUnit!
 		Method[] methods = TestDuckDBJDBC.class.getMethods();
+
+		Arrays.sort(methods, new Comparator<Method>() {
```

Comment:
For future reference, `Comparator.comparing(Method::getName)` is useful for code like this